src/components/select/select.js

```diff
@@ -202,13 +203,19 @@ function SelectDirective($mdSelect, $mdUtil, $mdTheming, $mdAria, $compile, $par
       var valueEl = element.find('md-select-value');
       var isReadonly = angular.isDefined(attr.readonly);
 
+      mdSelectCtrl.mdSelectHeaderInput = element.find('md-select-header').find('input')[0];
```

Comment:
I believe the intention of this directive (the header) is to let the user put in any custom content they desire. The multiple input check I do not believe makes sense in the context of a component header, but does make sense for the rest of the component.

[This upcoming feature](https://github.com/angular/material/issues/7385) is one use case where we would want a second input in the `md-select` directive that this check would clobber.